### PR TITLE
Fixes issue OP-58

### PR DESCRIPTION
**Pull request by rak16**
Bug Number: [OP-58](https://project-sunbird.atlassian.net/browse/OP-58)
Issue: Cloud search keeps showing the loading spinner even after the search has failed
RCA: When the search got triggered, a state variable got updated to indicate the search has begun which caused the spinner to get rendered. In case of an error the state variable was not being reset, thus causing the issue.
Fix: Reset the state variable in case of a failure.

On error:
screenshot from 2019-02-08 15-57-45

On clicking OK:
screenshot from 2019-02-08 15-57-55

### CODE REVIEW

File: devmgmtui/src/components/cloud/ContentArea.js [hunk 1]
```diff
@@ -148,10 +148,12 @@ function renderTable(content, handleDownload) {
 }
 
 function renderNoResultsFound(query) {
+	const placeholderText = !query ? "Nothing to show here" : `No results found for "${query}"`;
```
Comment (Skeletrox) [inline devmgmtui/src/components/cloud/ContentArea.js]: replace with

    query ? `No results found for "${query}"` : "Nothing to show here";

so as to avoid the unnecessary `NOT`ing of query

Comment (rak16) [reply devmgmtui/src/components/cloud/ContentArea.js]: Done

File: devmgmtui/src/components/cloud/CloudDownload.js [hunk 2]
```diff
@@ -176,6 +176,7 @@ class CloudDownload extends Component {
 				console.log(err);
 				
 				alert('Trouble fetching content from Diksha server.');
```
Comment (chetansk) [inline devmgmtui/src/components/cloud/CloudDownload.js]: This is not plugin code, you should not add "Diksha" in the alert message. Change it.

File: devmgmtui/src/components/cloud/CloudDownload.js [hunk 3]
```diff
@@ -320,8 +321,6 @@ class CloudDownload extends Component {
 
 	componentDidMount() {
 		document.title = 'Cloud Download';
-
```
Comment (chetansk) [inline devmgmtui/src/components/cloud/CloudDownload.js]: Why is this removed ?

File: devmgmtui/src/components/cloud/CloudDownload.js [hunk 4]
```diff
@@ -176,6 +176,7 @@ class CloudDownload extends Component {
 				console.log(err);
 				
 				alert('Trouble fetching content from Diksha server.');
+				this.props.clearCurrentContent();
```
Comment (chetansk) [inline devmgmtui/src/components/cloud/CloudDownload.js]: This doesn't need to be handled under different conditions, like failed download etc..

File: devmgmtui/src/components/cloud/CloudDownload.js (hunk 2 above)
Comment (rak16) [reply devmgmtui/src/components/cloud/CloudDownload.js]: The cloud endpoints are Diksha specific, thus the message. This is one of the many places where plugin specific code exists. Thought it would be best to decouple this when I am taking up that issue.

File: devmgmtui/src/components/cloud/CloudDownload.js (hunk 3 above)
Comment (rak16) [reply devmgmtui/src/components/cloud/CloudDownload.js]: It is stylistic. The space looked odd, nothing significant.

File: devmgmtui/src/components/cloud/CloudDownload.js (hunk 4 above)
Comment (rak16) [reply devmgmtui/src/components/cloud/CloudDownload.js]: Not really. This is specific to search results and no other related events require clearing of the state.
